File: JAVA/level1/시저_암호.java
class Solution {
  public String solution(String s, int n) {
      String answer = "";
      n %= 26;

      for(int i=0; i<s.length(); i++){
          if(s.charAt(i) == ' ')
              answer += " ";
          else{
              int index = (int)s.charAt(i) + n;
              if(s.charAt(i)>=65 && s.charAt(i)<=90){
                  if(index > 90)
                      index -= 26;
              }
              else{
                  if(index > 122)
                      index -= 26;
              }
              answer += (char)index;
          }
      }

      return answer;
  }
}

File: C++/level1/시저_암호.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string s, int n) {
    string answer = "";
    int index;
    int LUcase;
    
    for(auto c: s){
        if(c == ' '){
            answer += " ";
        }
        else{
            if(c >= 65 && c <= 90)
                LUcase = 1;
            else if(c >= 97 && c <= 122)
                LUcase = 2;
            
            c += n;
            
            if(LUcase == 1 && !(c >= 65 && c <= 90))
                c -= 26;
            if(LUcase == 2 && !(c >= 97 && c <= 122))
                c -= 26;

            answer += c;
        }
    }
    return answer;
}
